scripts/local-cluster.sh

```diff
@@ -0,0 +1,439 @@
+  - create <name> <config|flavor>                - creates a local cluster, optional args [ --skip-calico, --skip-minio ]
+  - delete <name>                                - deletes a local cluster
+  - list clusters                                - lists available clusters
+  - list flavors                                 - lists available flavors
+  "
+}
+log.continue() {
+  log.warn.no_newline "${1} [y/N]: "
+
+  read -r reply
+  if ! [[ "${reply}" =~ ^(y|Y|yes|Yes|YES)$ ]]; then
+    log.fatal "aborted"
+  fi
+}
+
+list.clusters() {
+  kind get clusters 2>/dev/null
+}
+list.flavors() {
+  echo "${HERE}/local-clusters/flavors/"*.yaml | tr ' ' '\n' | sed -e "s#${HERE}/local-clusters/flavors/##" -e 's#.yaml$##'
+}
+
+index.state() {
+  local cluster state
+  cluster="${1:-}"
+  state="${2:-}"
+
+  if [[ -z "${CK8S_CONFIG_PATH:-}" ]]; then
+    log.fatal "CK8S_CONFIG_PATH is unset"
+  fi
+
+  if ! [[ -d "${CK8S_CONFIG_PATH}" ]]; then
+    mkdir -p "${CK8S_CONFIG_PATH}"
+  fi
+
+  if ! [[ -f "${CK8S_CONFIG_PATH}/cluster-index.yaml" ]]; then
+    touch "${CK8S_CONFIG_PATH}/cluster-index.yaml"
+  fi
+
+  case "${state}" in
+  "")
+    yq4 ".\"${cluster}\"" "${CK8S_CONFIG_PATH}/cluster-index.yaml" ;;
+  "delete")
+    yq4 -i "del(.\"${cluster}\")" "${CK8S_CONFIG_PATH}/cluster-index.yaml" ;;
+  *)
+    yq4 -i ".\"${cluster}\" = \"${state}\"" "${CK8S_CONFIG_PATH}/cluster-index.yaml" ;;
+  esac
+}
+
+cluster.exist() {
+  kind get clusters 2>/dev/null | grep -E "^${1}$" &>/dev/null
+}
+
+config() {
+  local name flavor domain ops_prefix
+  name="${1:-}"
+  flavor="${2:-}"
+  domain="${3:-}"
+  ops_prefix="${4:-"ops"}"
+
+  if [[ -z "${name}" ]] || [[ -z "${flavor}" ]] || [[ -z "${domain}" ]]; then
+    log.usage
+  fi
+
+  if [[ -z "${CK8S_CONFIG_PATH:-}" ]]; then
+    log.fatal "CK8S_CONFIG_PATH is unset"
+  fi
+
+  if ! [[ -d "${CK8S_CONFIG_PATH}" ]]; then
+    mkdir -p "${CK8S_CONFIG_PATH}"
+  fi
+
+  if ! [[ -f "${CK8S_CONFIG_PATH}/common-config.yaml" ]]; then
+    touch "${CK8S_CONFIG_PATH}/common-config.yaml"
+  fi
+
+  if ! [[ -f "${CK8S_CONFIG_PATH}/sc-config.yaml" ]]; then
+    touch "${CK8S_CONFIG_PATH}/sc-config.yaml"
+  fi
+
+  if ! [[ -f "${CK8S_CONFIG_PATH}/wc-config.yaml" ]]; then
+    touch "${CK8S_CONFIG_PATH}/wc-config.yaml"
+  fi
+
+  print_coredns_config() {
+    cat <<EOF
+${domain}:53 {
+  errors
+  bind 169.254.20.10 10.96.0.10
+  template IN A ${domain} {
+    match \"\.${domain//\./\\\.}\.$\"
+    answer \"{{ .Name }} 60 IN A 10.96.0.20\"
+    fallthrough
+  }
+  cache 30
+  reload
+  loop
+  forward . 1.1.1.1 1.0.0.1
+}
+.:30053 {
+  errors
+  log
+  template IN A ${domain} {
+    match \"\.${domain//\./\\\.}\.$\"
+    answer \"{{ .Name }} 60 IN A 127.0.64.43\"
+    fallthrough
+  }
+  cache 30
+  reload
+  loop
+  forward . 1.1.1.1 1.0.0.1
+}
+EOF
+  }
+
+  yq4 -Pi 'with(select(. == null); . = {}) | . *= {
+    "global": {
+      "baseDomain": "'"${domain}"'",
+      "opsDomain": "'"${ops_prefix}.${domain}"'",
```

Comment:
These could be passed without string interpolation using [`strenv()`](https://mikefarah.gitbook.io/yq/operators/env-variable-operators) like

```bash
domain="{domain}" yq4 -Pi '.global.domain = strenv(domain)`
```

Reply:
Well this is a game changer.